src/lib/x509/x509self.cpp

```diff
@@ -21,25 +21,52 @@ namespace {
 /*
 * Load information from the X509_Cert_Options
 */
-void load_info(const X509_Cert_Options& opts, X509_DN& subject_dn, AlternativeName& subject_alt) {
+X509_DN load_dn_info(const X509_Cert_Options& opts) {
+   X509_DN subject_dn;
+
    subject_dn.add_attribute("X520.CommonName", opts.common_name);
    subject_dn.add_attribute("X520.Country", opts.country);
    subject_dn.add_attribute("X520.State", opts.state);
    subject_dn.add_attribute("X520.Locality", opts.locality);
    subject_dn.add_attribute("X520.Organization", opts.organization);
    subject_dn.add_attribute("X520.OrganizationalUnit", opts.org_unit);
+   subject_dn.add_attribute("X520.SerialNumber", opts.serial_number);
+
    for(const auto& extra_ou : opts.more_org_units) {
       subject_dn.add_attribute("X520.OrganizationalUnit", extra_ou);
    }
 
-   subject_dn.add_attribute("X520.SerialNumber", opts.serial_number);
-   subject_alt = AlternativeName(opts.email, opts.uri, opts.dns, opts.ip);
+   return subject_dn;
+}
+
+void load_alt_name(const X509_Cert_Options& opts, AlternativeName& subject_alt) {
+   subject_alt.add_attribute("DNS", opts.dns);
+   subject_alt.add_attribute("URI", opts.uri);
+   subject_alt.add_attribute("RFC822", opts.email);
+   subject_alt.add_attribute("IP", opts.ip);
    subject_alt.add_othername(OID::from_string("PKIX.XMPPAddr"), opts.xmpp, ASN1_Type::Utf8String);
 
    for(const auto& dns : opts.more_dns) {
       subject_alt.add_attribute("DNS", dns);
    }
 }
+
+void create_alt_name_ext(const X509_Cert_Options& opts, Extensions& extensions) {
+   /*
+   If the extension was already created in opts.extension we need to
+   merge that with any values that were set in the other fields
+   */
+   if(auto ext = extensions.get_extension_object_as<Cert_Extension::Subject_Alternative_Name>()) {
+      AlternativeName subject_alt = ext->get_alt_name();
+      load_alt_name(opts, subject_alt);
+      extensions.replace(std::make_unique<Cert_Extension::Subject_Alternative_Name>(subject_alt));
+   } else {
+      AlternativeName subject_alt;
+      load_alt_name(opts, subject_alt);
+      extensions.replace(std::make_unique<Cert_Extension::Subject_Alternative_Name>(subject_alt));
+   }
+}
```

Comment:
Maybe remove the partly redundant if-statement by something like.
```cpp
AlternativeName subject_alt = [&]{
   /*
      If the extension was already created in opts.extension we need to
      merge that with any values that were set in the other fields
   */
   if(auto ext = 
   extensions.get_extension_object_as<Cert_Extension::Subject_Alternative_Name>()) {
      return ext->get_alt_name();
   } else {
      return AlternativeName(); // new
   }
}();
load_alt_name(opts, subject_alt);
extensions.replace(std::make_unique<Cert_Extension::Subject_Alternative_Name>(subject_alt));

```

Reply:
I realized we can just create an empty one then assign to it if the extension is set, much cleaner